posts/2021-05-27-incremental-progress.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+title: "Incremental Compilation: Making a Comeback"
+author: Felix Klock
+team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+On May 10th, we [released Rust 1.52.1][1.52.1 blog post], an emergency point
+release working around incremental compiler bugs that had been promoted into
+compiler errors. This post provides a status update on what we have done since
+then, and ends by recommending Rust developers try out the 1.53-beta release:
+Doing so benefits everyone, especially you.
+
+The changes in 1.52.1 over 1.52.0 were:
+ 1. We made incremental compilation
+something that developers had to opt into, via a `RUSTC_FORCE_INCREMENTAL`
+environment variable, so that developers would not encounter the compiler error by accident, and
+ 2. We improved the
+diagnostic messages for the newly added error, so that developers who *did* opt into using incremental compilation could effectively respond
+if they ever encountered the error.
+
+The [1.52.1 blog post][] explains in more detail what the bugs were and why we
+took the step of making incremental-compilation an opt-in feature for that
+release. The short version of the story is: A new internal "fingerprint" check
+had revealed lots of (old) inconsistencies within the incremental-compilation
+infrastructure, and at least one of those inconsistencies could lead to
+miscompilations.
+
+[1.52.1 blog post]: https://blog.rust-lang.org/2021/05/10/Rust-1.52.1.html
+
+## Incremental Progress
+
+Since then, we have made a lot of progress addressing all of these
+inconsistencies. The internal "fingerprint" check, combined with all the great
+bug reports supplied by Rust developers like you, gave us the information we
+needed to fix the serious bugs related to this issue.
+
+If you want to get more details about those bugs, and how each has been fixed, I
+recommend seeing the summary [issue #84970][].
+
+[issue #84970]: https://github.com/rust-lang/rust/issues/84970
+
+I am proud to report that we have now backported the relevant fixes to the 1.53
+beta channel. Therefore, starting with the 1.53.0 release, we will resume
+providing incremental-compilation support *without* requiring use of the
+`RUSTC_FORCE_INCREMENTAL` environment variable.
+
+## 1.53-beta: Not Just For βreakfast Anymore
```

Comment:
```suggestion
## 1.53-beta: Not Just For Breakfast Anymore
```
This makes it more readable with screenreaders.

Reply:
Oh, that's a good point. I should probably change the "βetta" down below to "Betta" too.